src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1250,10 +1221,14 @@ bool is_object_superset(
         // newer)
         return false;
     }
-    if (!is_object_pattern_properties_superset(ctx, older, newer)) {
-        // pattern properties checks are not compatible
-        return false;
-    }
+
+    // note: to match the behavior of the legacy software,
+    // ```
+    // if(!is_object_pattern_properties_superset(ctx, older, newer))
+    //   return false;
+    // ```
+    // is omitted
+
```

Comment:
nitpick: this comment is just reproducing the diff, in a sense, so probably not very high value unless we're planning on switching it on or something.

Reply:
i felt sad completely removing a piece of my work :D